module/os/linux/zfs/zfs_uio.c

```diff
@@ -633,7 +649,10 @@ zfs_uio_get_dio_pages_alloc(zfs_uio_t *uio, zfs_uio_rw_t rw)
 
 	if (error) {
 #if defined(HAVE_PIN_USER_PAGES_UNLOCKED)
-		unpin_user_pages(uio->uio_dio.pages, uio->uio_dio.npages);
+		if (uio->uio_dio.pinned)
+			unpin_user_pages(uio->uio_dio.pages,
+			    uio->uio_dio.npages);
+		else
 #else
```

Comment:
Same as above, here should be `#endif` instead of `#else`.

Reply:
Yep, that as a complete brain fart on my part. I updated this.